packages/use-contractkit/src/connectors/connectors.ts

```diff
@@ -39,6 +40,70 @@ export class UnauthenticatedConnector implements Connector {
   }
 }
 
+/* eslint-disable */
+// TODO: Move this to wallet-base maybe
+class ReadOnlyWallet implements Wallet {
+  constructor(public readonly account: string) {}
+  addAccount() {}
+  getAccounts() {
+    return [this.account];
+  }
+  removeAccount() {}
+  hasAccount(address?: string | undefined) {
+    return !!address && address === this.account;
+  }
+  // @ts-ignore
+  signTransaction() {
+    throw Error('Not implemented');
+  }
+  // @ts-ignore
+  signTypedData() {
+    throw Error('Not implemented');
+  }
+  // @ts-ignore
+  signPersonalMessage() {
+    throw Error('Not implemented');
+  }
+  // @ts-ignore
+  decrypt() {
+    throw Error('Not implemented');
+  }
+  // @ts-ignore
+  computeSharedSecret() {
+    throw Error('Not implemented');
+  }
+}
+/* eslint-enable */
+
+export class ReadOnlyConnector implements Connector {
+  public initialised = true;
+  public type = WalletTypes.ReadOnly;
+  public kit: ContractKit;
+  constructor(n: Network, public readonly account: string) {
+    localStorage.setItem(
+      localStorageKeys.lastUsedWalletType,
+      WalletTypes.ReadOnly
+    );
+    localStorage.setItem(
+      localStorageKeys.lastUsedWalletArguments,
+      JSON.stringify([account])
+    );
+    localStorage.setItem(localStorageKeys.lastUsedNetwork, n.name);
+    // eslint-disable-next-line
+    // @ts-ignore
```

Comment:
ditto. sorry for nitpicking but it's helpful to leave a comment why you're disabling/ignoring

Reply:
7ca1cbe